### PR TITLE
Move reset badge dismiss code to server, more teams cleanup

### PR DESCRIPTION
**Pull request by jakob223**

### CODE REVIEW

**Comment by jakob223** [comment]
see server PR https://github.com/keybase/keybase/pull/4996
cc @keybase/y2ksquad @mlsteele

**Comment by jakob223** [comment]
Hmm, this cleanup was premature, it breaks a bunch of things until the new team reload strategy is finished.

**Comment by jakob223** [comment]
@keybase/react-hackers this is re-ready for review. As you can see from the PR, I had a fight with `git` and lost at some point during this work; I'm trying to fix the history right now but you can review the first 5ish commits and the last one.

**Comment by jakob223** [comment]
Scratch that, it's just two commits: https://github.com/keybase/client/pull/21961/commits/5f7d313f8a7770d1ca839c8cae7e80889c77fb99 and https://github.com/keybase/client/pull/21961/commits/fd3d8c9f36e9f0a3753887d44a8079143e61f342

File: shared/actions/teams.tsx [hunk 1]
```diff
@@ -483,10 +483,11 @@ const ignoreRequest = async (action: TeamsGen.IgnoreRequestPayload) => {
       {name: teamname, username},
       Constants.teamWaitingKey(teamname)
     )
-    return TeamsGen.createGetDetails({teamname})
+    return []
```
Comment (chrisnojima) [inline shared/actions/teams.tsx]: should return false

File: shared/actions/teams.tsx [hunk 2]
```diff
@@ -601,6 +606,16 @@ function* getDetails(state: TypedState, action: TeamsGen.GetDetailsPayload, logg
   }
 }
 
+function* getDetails(state: TypedState, action: TeamsGen.GetDetailsPayload) {
```
Comment (chrisnojima) [inline shared/actions/teams.tsx]: doesn't need to be a genrator

File: shared/teams/subscriber.tsx [hunk 3]
```diff
@@ -34,12 +35,51 @@ export const useTeamsSubscribe = Container.isMobile ? useTeamsSubscribeMobile :
...
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => () => dispatch(TeamsGen.createUnsubscribeTeamList()), [])
+}
+const useTeamDetailsSubscribeDesktop = (teamID: Types.TeamID) => {
+  const dispatch = Container.useDispatch()
+  React.useEffect(() => {
+    dispatch(TeamsGen.createGetDetailsByID({_subscribe: true, teamID}))
+    return () => dispatch(TeamsGen.createUnsubscribeTeamDetails({teamID}))
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```
Comment (chrisnojima) [inline shared/teams/subscriber.tsx]: you probably do want teamID/dispatch in the list

**Comment by jakob223** [comment]
CI failure before merge was broken in master, merging

**Pull request merged**